src/OpenTelemetry/Metrics/DataPoint/DataValue.cs

```diff
@@ -0,0 +1,48 @@
+// <copyright file="DataValue.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+namespace OpenTelemetry.Metrics
+{
+    public readonly struct DataValue : IDataValue
+    {
+        private readonly IDataValue value;
+
+        internal DataValue(int value)
+        {
+            // Promote to long
+            this.value = new DataValue<int>(value);
```

Comment:
Just double-checking, did you mean to do `DataValue<long>`?

Reply:
Yes. My typo.